### PR TITLE
ARROW-15697: [R] Add logo and meta tags to pkgdown site

### PR DESCRIPTION
**Pull request by djnavarro**
A minor PR focused on the pkgdown site:

- adds the hex sticker to `README.md` (and home page of the R pkgdown site)
- uses the nicely-spaced horizontal Apache Arrow logo as the social media card

### CODE REVIEW

**Comment by ianmcook** [comment]
I believe we've had some trouble keeping the R package small enough to be accepted by CRAN without NOTEs. This PR adds about 100 KB of PNG files which might worsen this. Is it possible to use SVGs?

**Comment by djnavarro** [comment]
As far as I am aware, twitter doesn't accept SVG files as social media previews. What I've done instead is remove the hex sticker entirely (that isn't really needed to make the meta tags work it's just there to make the README a little prettier), and added the logo file to .Rbuildignore. I think that should ensure that the PNG logo will be included in the pkgdown site (and hence appear on twitter) but not the CRAN submission?

File: r/_pkgdown.yml [hunk 1]
```diff
@@ -23,6 +23,14 @@ template:
   params:
     bootswatch: cosmo
     ganalytics: UA-107500873-1
+  opengraph:
+    image:
+      src: man/figures/card.png
```
Comment (nealrichardson) [inline r/_pkgdown.yml:28]: I forget, does this work? If so, you don't need to put a copy here at all.

```suggestion
      src: https://arrow.apache.org/img/arrow-logo_horizontal_black-txt_white-bg.png
```

**Review by nealrichardson — COMMENTED** [review]
One possible suggestion, but if that's no good, I think this is fine. It's not a huge file. 

Another option, if we didn't want to check in a .png here and we can't just reference it by URL elsewhere, is to download this image in the build script where we generate the docs (around [here](https://github.com/apache/arrow/blob/master/ci/scripts/r_build.sh#L34)). That makes it more fragile but avoids checking in a binary blob to the repo, which I know we try to avoid (though this one is small so it's not a big deal).

File: r/_pkgdown.yml (hunk 1 above)
Comment (djnavarro) [reply r/_pkgdown.yml:28]: 🤦🏻‍♀️  why didn't I think of that? Yes, that's much easier! 😁

**Review by jonkeane — APPROVED** [review]

**Pull request closed without merging**